var/spack/repos/builtin.mock/packages/mpich/package.py

```diff
@@ -4,6 +4,9 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 from spack.package import *
+from spack.util.executable import Executable
+
+ninja = Executable("ninja")  # to appease ninja test
```

Comment:
Can you explain what this is doing?

Reply:
All of the ninja build_system tests assume that a ninja `Executable` object exists in the module scope of this mock package.  Adding it as a dependency doesn't work, because it can't actually build it, so the build_system tests need us to find the ninja executable in the environment.  This was my initial stopgap for keeping them all working, but it doesn't make me particularly happy.

Reply:
Had a bit more of a think about this and reworked the ninja handling, which took some tweaking of `which` and a couple other spots, but I think puts us in a much better place.  Apparently we weren't actually setting `-j` reliably on ninja unless it was being set by the caller, that's fixed, also put it back in the default environment if it's available because it really is a bottom dependency, especially on windows.  Bit more churn, but I like the result a whole lot more.